packages/beacon-node/src/api/impl/beacon/blocks/index.ts

```diff
@@ -297,17 +299,32 @@ export function getBeaconBlockApi({
     const source = ProducedBlockSource.builder;
     chain.logger.debug("Reconstructing  signedBlockOrContents", {slot, blockRoot, source});
 
-    const signedBlockOrContents = await reconstructBuilderBlockOrContents(chain, {
-      data: signedBlindedBlock,
-      bytes: context?.sszBytes,
-    });
+    let signedBlockOrContents: SignedBeaconBlockOrContents | undefined = undefined;
 
-    // the full block is published by relay and it's possible that the block is already known to us
-    // by gossip
-    //
-    // see: https://github.com/ChainSafe/lodestar/issues/5404
-    chain.logger.info("Publishing assembled block", {slot, blockRoot, source});
-    return publishBlock({signedBlockOrContents}, {...context, sszBytes: null}, {...opts, ignoreIfKnown: true});
+    if (isForkPostFulu(fork)) {
+      chain.logger.info("Submit blinded block to builder for publishing", {slot, blockRoot});
+      await delegateBlindedBlockToBuilder(chain, {
+        data: signedBlindedBlock,
+        bytes: context?.sszBytes,
+      });
```

Comment:
I would rather emit the log after submitting the block
```suggestion
      await delegateBlindedBlockToBuilder(chain, {
        data: signedBlindedBlock,
        bytes: context?.sszBytes,
      });
      chain.logger.info("Submitted blinded block to builder for publishing", {slot, blockRoot});
```